test/unit/org/apache/cassandra/index/sai/disk/v1/bbtree/BlockBalancedTreeReaderTest.java

```diff
@@ -222,6 +231,55 @@ public void testResourcesReleaseWhenQueryDoesntMatchAnything() throws Exception
         assertNull(intersection);
     }
 
+    @Test
+    public void testConcurrentIntersectionsOnSameReader() throws Exception
+    {
+        int numRows = 1000;
+
+        final BlockBalancedTreeRamBuffer buffer = new BlockBalancedTreeRamBuffer(Integer.BYTES);
+
+        byte[] scratch = new byte[4];
+        for (int docID = 0; docID < numRows; docID++)
+        {
+            NumericUtils.intToSortableBytes(docID, scratch, 0);
+            buffer.add(docID, scratch);
+        }
+
+        final BlockBalancedTreeReader reader = finishAndOpenReader(4, buffer);
```

Comment:
nit: Little hygiene thing, but I guess we should make sure we close the reader here.

(I can fix on commit.)